app/init.php

```diff
@@ -1942,3 +1943,13 @@ function getDevice(string $root, string $connection = ''): Device
 
     return '';
 }, ['request']);
+
+App::setResource('apiKey', function (Request $request, Document $project): ?Key {
+    $key = $request->getHeader('x-appwrite-key');
+
+    if (empty($key)) {
+        return null;
+    }
```

Comment:
Let's make `Key::decode` handle this case, so it returns guestKey instad. This prevents `null` from ever hapenning, and makes later use much safer

Reply:
I don't think it makes sense here, `null` is the absence of a key, which is different to the guest key. This way, later checks are just "is there a key" vs "is the key a guest key"